src/analysis/seminorm.lean

```diff
@@ -771,9 +771,24 @@ begin
     rw seminorm.mem_ball_zero at hy,
     exact (mul_lt_mul_left hk).mpr hy },
   refine ⟨k⁻¹ • x, _, _⟩,
-  { rw [seminorm.mem_ball_zero, map_smul_eq_mul, norm_inv, ←(mul_lt_mul_left hk),
-      ←mul_assoc, ←(div_eq_mul_inv ∥k∥ ∥k∥), div_self (ne_of_gt hk), one_mul],
-    exact h},
+  { rwa [seminorm.mem_ball_zero, map_smul_eq_mul, norm_inv, ←(mul_lt_mul_left hk),
+      ←mul_assoc, ←(div_eq_mul_inv ∥k∥ ∥k∥), div_self (ne_of_gt hk), one_mul] },
+  rw [←smul_assoc, smul_eq_mul, ←div_eq_mul_inv, div_self (norm_pos_iff.mp hk), one_smul],
+end
+
+lemma smul_closed_ball_zero {p : seminorm 𝕜 E} {k : 𝕜} {r : ℝ} (hk : 0 < ∥k∥) :
+  k • p.closed_ball 0 r = p.closed_ball 0 (∥k∥ * r) :=
```

Comment:
What are your thoughts on extracting a `k • p.closed_ball 0 r \le p.closed_ball 0 (∥k∥ * r)` lemma, which would be true without `hk`? I don't know how useful it would be, but it seems like an obvious way to split the proof below.

Reply:
That makes sense, I'll do that.

Reply:
Actually this direction also needs the `hk` hypothesis. `k=0` and `r>0` gives a counterexample, because the RHS is empty and the LHS is `{0}`.

Reply:
Is it possible I meant to say that the other way around, or do both directions need the hypothesis?

Reply:
Right, I can make the other one work.

Reply:
Oh actually I just realized I was looking at the open ball version :facepalm: So it looks like what you originally suggested does work for closed ball (but not the converse!), and the situation is reversed for open balls!